src/util/power.cpp

```diff
@@ -0,0 +1,32 @@
+#include <fstream>
+
+#include "power.hpp"
+
+namespace nest {
+namespace mc {
+namespace util {
+
+#ifdef NMC_HAVE_CRAY
+
+energy_size_type energy() {
+    energy_size_type result = -1;
+
+    std::ifstream fid("/sys/cray/pm_counters/energy");
+    if (fid.good()) {
```

Comment:
```
If extraction fails, zero is written to value and failbit is set. If extraction results in the value too large or too small to fit in value, std::numeric_limits<T>::max() or std::numeric_limits<T>::min()  is written and failbit flag is set. (since C++11)
```